### PR TITLE
Analyzer: Do not resolve remote table id on initiator

### PR DESCRIPTION
**Pull request by novikd**
### Changelog category (leave one):

- Not for changelog (changelog entry is not required)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Do not resolve table_id on the initiator. There could be no such a table on the initiator.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 4e5249275ed67c52d958007978c66619db22a1a5 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59073/4e5249275ed67c52d958007978c66619db22a1a5/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59073/4e5249275ed67c52d958007978c66619db22a1a5/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Comment by qoega** [comment]
Can we add a test? It was not caught automatically

**Comment by novikd** [comment]
> Can we add a test? It was not caught automatically

That's a good question. I haven't come up with something good yet.

File: src/Storages/StorageDistributed.cpp [hunk 1]
```diff
@@ -779,18 +779,11 @@ QueryTreeNodePtr buildQueryTreeDistributed(SelectQueryInfo & query_info,
     }
     else
     {
-        auto resolved_remote_storage_id = remote_storage_id;
-        // In case of cross-replication we don't know what database is used for the table.
-        // `storage_id.hasDatabase()` can return false only on the initiator node.
-        // Each shard will use the default database (in the case of cross-replication shards may have different defaults).
-        if (remote_storage_id.hasDatabase())
-            resolved_remote_storage_id = query_context->resolveStorageID(remote_storage_id);
```
Comment (nickitat) [inline src/Storages/StorageDistributed.cpp:787]: > There could be no such a table on the initiator

maybe we could detect such a case and only then ignore resolving result? this code seems to be added for a reason and we prob better to keep it.

File: tests/queries/0_stateless/02971_analyzer_remote_id.sh [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+CURDIR=$(cd "$(dirname "${BASH_SOURCE[0]}")" && pwd)
+# shellcheck source=../shell_config.sh
+. "$CURDIR"/../shell_config.sh
+
+${CLICKHOUSE_CLIENT} --query="DROP DATABASE IF EXISTS test_02971"
+${CLICKHOUSE_CLIENT} --query="CREATE DATABASE test_02971"
+
+${CLICKHOUSE_CLIENT} --query="CREATE TABLE test_02971.x ENGINE = MergeTree() ORDER BY number AS SELECT * FROM numbers(2)"
+${CLICKHOUSE_LOCAL} --query="SELECT count() FROM remote('127.0.0.{2,3}', 'test_02971.x') SETTINGS allow_experimental_analyzer = 1"
```
Comment (nickitat) [inline tests/queries/0_stateless/02971_analyzer_remote_id.sh:11]: do you avoid querying from `127.0.0.1` intentionally here or it doesn't matter?

Comment (novikd) [reply tests/queries/0_stateless/02971_analyzer_remote_id.sh:11]: It doesn't matter. I just simulate an initiator and 2 separate shards.

**Review by nickitat — APPROVED** [review]

**Pull request merged**